code/game/objects/items.dm

```diff
@@ -561,8 +639,21 @@ var/list/global/slot_flags_enumeration = list(
 		if(blood_DNA[M.dna.unique_enzymes])
 			return 0 //already bloodied with this blood. Cannot add more.
 		blood_DNA[M.dna.unique_enzymes] = M.dna.b_type
+	is_bloodied = 1
+	was_bloodied = 1
+	update_icon()
 	return 1 //we applied blood to the item
 
+/obj/item/proc/update_blood()//For when luminol is sprayed on an object with still-visible blood, this will fix the colour
+	if( !blood_overlay )
+		generate_blood_overlay()//make sure the blood overlay is initialised
+
+	for (var/V in overlays)//Make sure that the overlay is already visible
+		var/image/I = V
+		if (I.name == "blood")

```

Comment:
I assume this has been tested? It goes against the convention of dealing with overlays, whereby the image is saved as a variable on the object itself, and then removed in a much simpler fashion.